### PR TITLE
fix(chips): Fix incorrect ripple effect on filter chip

### LINKED ISSUE #2812: Incorrect ripple effect on filter chip with no leading icon
## Bugs

The filter chip with no leading icon has incorrect ripple when clicked. Instead of starting the ripple where clicked the ripple center is always in the same spot.

### What MDC Web Version are you using?

> 0.35.2

### What browser(s) is this bug affecting?

> All

### What OS are you using?

> Mac OS

### What are the steps to reproduce the bug?

> 1. Run the demo server
> 2. Go to localhost:8080/chips.html
> 3. Click on a Filter chip with no leading icon 
> 4. Observe the component's behavior
> 5. Helps to see if you set animations to 10%

### What is the expected behavior?

> The ripple should start where the mouse is clicked on the chip. 

### What is the actual behavior?

> The ripple effect is starting in the same spot and not where the chip is clicked.

### Any other information you believe would be useful?

Incorrect behavior
bad-chip

Correct behavior
correct-ripple

### PR DESCRIPTION
**Pull request by TroyTae**
Fixes #2812

### CODE REVIEW

File: packages/mdc-chips/chip/foundation.ts [hunk 1]
```diff
@@ -110,7 +110,18 @@ export class MDCChipFoundation extends MDCFoundation<MDCChipAdapter> {
         // when adding it to the root client rect's width.
         const checkmarkWidth = checkmarkRect.height;
         const width = rootRect.width + checkmarkWidth;
-        return {...rootRect, width, height};
+        // Spread expression translate to Object.assign in Typescript.
```
Comment (abhiomkar) [inline packages/mdc-chips/chip/foundation.ts]: Good catch!

You can directly return the object with calculated height & width, so we don't need to use spread operator anymore.

```
return rect = {
  bottom: rootRect.bottom,
  height: rootRect.height,
  left: rootRect.left,
  right: rootRect.right,
  top: rootRect.top,
  width: width,
};
```

Comment (abhiomkar) [inline packages/mdc-chips/chip/foundation.ts]: We don't need this line anymore

```
const height = rootRect.height;
```

Comment (TroyTae) [reply packages/mdc-chips/chip/foundation.ts]: Is there any special meaning this syntax? ```return rect = {};```
I guess you wanna talk this: ```return {}```
So I reflect it :)

Comment (TroyTae) [reply packages/mdc-chips/chip/foundation.ts]: Thank you for your review!

Comment (abhiomkar) [reply packages/mdc-chips/chip/foundation.ts]: Sorry for the typo. Yes I meant `return {}`.

**Pull request merged**